src/AutoRest.CSharp.V3/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -55,9 +55,20 @@ public static void WriteParameterNullChecks(this CodeWriter writer, IReadOnlyCol
         {
             foreach (Parameter parameter in parameters)
             {
-                if (parameter.DefaultValue != null && !CanBeInitializedInline(parameter))
+                if (parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue))
                 {
-                    writer.Line($"{parameter.Name} ??= new {parameter.Type}({parameter.DefaultValue.Value.Value:L});");
+                    if (TypeFactory.IsStruct(parameter.Type))
+                    {
+                        writer.Append($"{parameter.Name} ??= ");
+                        WriteConstant(writer, parameter.DefaultValue.Value);
+                    }
+                    else
+                    {
+                        writer.Append($"{parameter.Name} ??= new {parameter.Type}(");
```

Comment:
Can we move the `new {parameter.Type}` part into WriteConstant too?

Reply:
I tried to do that but when I create an object of `Constant` it throws because the value type(string type) and framework type(Uri type) are different. https://github.com/ShivangiReja/autorest.csharp/blob/Generate_Content.json/src/AutoRest.CSharp.V3/Output/Models/Shared/Constant.cs#L43

Reply:
Aha, cool, let's leave this for the next time.